src/transformers/models/canine/tokenization_canine.py

```diff
@@ -0,0 +1,239 @@
+
+    Refer to superclass :class:`~transformers.PreTrainedTokenizer` for usage examples and documentation concerning
+    parameters.
+
+    Args:
+        model_max_length (:obj:`int`, `optional`, defaults to 2048):
+                The maximum sentence length the model accepts.
+    """
+
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+
+    def __init__(
+        self,
+        bos_token="[CLS]",
+        eos_token="[SEP]",
+        sep_token="[SEP]",
+        cls_token="[CLS]",
+        pad_token="[PAD]",
+        mask_token="[MASK]",
+        add_prefix_space=False,
+        model_max_length=2048,
+        **kwargs
+    ):
+        bos_token = AddedToken(bos_token, lstrip=False, rstrip=False) if isinstance(bos_token, str) else bos_token
+        eos_token = AddedToken(eos_token, lstrip=False, rstrip=False) if isinstance(eos_token, str) else eos_token
+        sep_token = AddedToken(sep_token, lstrip=False, rstrip=False) if isinstance(sep_token, str) else sep_token
+        cls_token = AddedToken(cls_token, lstrip=False, rstrip=False) if isinstance(cls_token, str) else cls_token
+        pad_token = AddedToken(pad_token, lstrip=False, rstrip=False) if isinstance(pad_token, str) else pad_token
+
+        # Mask token behave like a normal word, i.e. include the space before it
+        mask_token = AddedToken(mask_token, lstrip=True, rstrip=False) if isinstance(mask_token, str) else mask_token
+
+        super().__init__(
+            bos_token=bos_token,
+            eos_token=eos_token,
+            sep_token=sep_token,
+            cls_token=cls_token,
+            pad_token=pad_token,
+            mask_token=mask_token,
+            add_prefix_space=add_prefix_space,
+            model_max_length=model_max_length,
+            **kwargs,
+        )
+
+        # Creates a mapping for looking up the IDs of special symbols.
+        self._special_codepoints: Dict[Text, int] = {}
+        for codepoint, name in SPECIAL_CODEPOINTS.items():
+            self._special_codepoints[name] = codepoint
+
+        # Creates a mapping for looking up the string forms of special symbol IDs.
+        self._special_codepoint_strings: Dict[int, Text] = {
+            codepoint: name for name, codepoint in self._special_codepoints.items()
+        }
+
+        self._unicode_vocab_size = 1114112  # Unicode defines 1,114,112 total “codepoints”
+        self._num_special_tokens = len(self._special_codepoints)
+
+    @property
+    def vocab_size(self):
+        return self._unicode_vocab_size + self._num_special_tokens
+
+    def _tokenize(self, text: str) -> List[str]:
+        """Tokenize a string (i.e. perform character splitting)."""
+
+        pieces = [x for x in text]
+
+        return pieces
+
+    def _convert_token_to_id(self, token: str):
+        """Converts a token (i.e. a Unicode character) in an id (i.e. its integer Unicode code point value)."""
+
+        if token in self._special_codepoints:
```

Comment:
This shouldn't be needed because the special codepoints are all, by definiton, codepoints, and therefore `ord` will just do the right thing.

Reply:
Indeed, will fix this.